operator/v1/types_network.go

```diff
@@ -561,7 +563,35 @@ type IPsecConfig struct {
 	// this is left to the user to configure.
 	// +kubebuilder:validation:Enum=Disabled;External;Full
 	// +optional
+	// +unionDiscriminator
 	Mode IPsecMode `json:"mode,omitempty"`
+
+	// full defines configuration parameters for the IPsec `Full` mode.
+	// This is permitted only when mode is configured with `Full`,
+	// and forbidden otherwise.
+	// +unionMember,optional
+	// +optional
+	Full *IPsecFullModeConfig `json:"full,omitempty"`
+}
+
+type Encapsulation string
+
+const (
+	// ForceEncapsulation forces UDP encapsulation regardless of whether NAT is detected.
+	ForceEncapsulation = "Force"
+	// DisableEncapsulation disables UDP encapsulation even if NAT is present.
+	DisableEncapsulation = "Disable"
+	// AutoEncapsulation determines use of UDP encapsulation based on the detection of NAT.
+	AutoEncapsulation = "Auto"
+)
+
+type IPsecFullModeConfig struct {
+	// encapsulation option to configure libreswan on how inter-pod traffic across nodes
+	// are encapsulated to handle NAT traversal. When configured it uses UDP port 4500
+	// for the encapsulation.
+	// +kubebuilder:validation:Enum:=Force;Disable;Auto;""
+	// +optional
+	Encap Encapsulation `json:"encapsulation,omitempty"`
```

Comment:
Any particular reason why the struct field name is `Encap` and not `Encapsulation`? Generally, I would recommend having the Go struct field name match the JSON field name where possible.

Reply:
ok, changed it.